pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -242,21 +260,31 @@ func (l *ListOptionIndexer) deleteIndexFields(key string, tx db.TXClient) error
 	return nil
 }
 
-func (l *ListOptionIndexer) deleteLabels(key string, tx db.TXClient) error {
+func (l *ListOptionIndexer) deleteLabels(key string, _ any, tx db.TXClient) error {
 	err := tx.StmtExec(tx.Stmt(l.deleteLabelsStmt), key)
 	if err != nil {
 		return &db.QueryError{QueryString: l.deleteLabelsQuery, Err: err}
 	}
 	return nil
 }
 
+func (l *ListOptionIndexer) GetLastResourceVersion() string {
+	return l.resourceVersionCache.getLatest()
+}
+
 // ListByOptions returns objects according to the specified list options and partitions.
 // Specifically:
 //   - an unstructured list of resources belonging to any of the specified partitions
 //   - the total number of resources (returned list might be a subset depending on pagination options in lo)
 //   - a continue token, if there are more pages after the returned one
 //   - an error instead of all of the above if anything went wrong
 func (l *ListOptionIndexer) ListByOptions(ctx context.Context, lo ListOptions, partitions []partition.Partition, namespace string) (*unstructured.UnstructuredList, int, string, error) {
+	isMaybeStale := lo.Revision != "" && !l.resourceVersionCache.contains(lo.Revision)
+	if isMaybeStale {
```

Comment:
I don't get how we can "maybe" return an error message, but can live with this for now.